### PR TITLE
Define a language in REST context

### PR DESCRIPTION
**Pull request by Screenfeed**
Whenever a `lang` key is sent in `$_REQUEST`, define a current language as a `PLL_REST_Request->curlang` public property, like in other contexts.
The property can be an instance of `PLL_Language`, or `false` if something goes wrong, or `null` if no `lang` arg is sent.
Fixes [PLL Pro 949](https://github.com/polylang/polylang-pro/issues/949).

### CODE REVIEW

**Review by manooweb — APPROVED** [review]
Ok 👍
Notice that:
- add scripts in composer.json could be in conflict with the PR #898 
- the page list block is correctly filtered by the REST request language when it is added however when we change the language in the metabox the block isn't filtered by the new language because the block render isn't rerun when the post is saved. We need to refresh the admin edit post page.

**Comment by Screenfeed** [comment]
@manooweb Should I remove the composer scripts from this PR? They are useless now with #898 and may prevent a conflict.

**Comment by Screenfeed** [comment]
Done.

**Review by manooweb — APPROVED** [review]

File: include/rest-request.php [hunk 1]
```diff
@@ -50,23 +55,41 @@ class PLL_REST_Request extends PLL_Base {
...
-				$this->static_pages = new PLL_Static_Pages( $this );
+			if ( empty( $this->curlang ) && ! empty( $this->options['default_lang'] ) && is_string( $this->options['default_lang'] ) ) {
+				// A lang has been requested but it is invalid, let's fall back to the default one.
+				$this->curlang = $this->model->get_language( sanitize_key( $this->options['default_lang'] ) );
 			}
+		}
 
-			$this->links = new PLL_Admin_Links( $this );
+		if ( ! empty( $this->curlang ) ) {
+			$GLOBALS['text_direction'] = $this->curlang->is_rtl ? 'rtl' : 'ltr';
```
Comment (Chouby) [inline include/rest-request.php:74]: As we don't flter the locale, we should not change the text direction.

**Review by Chouby — CHANGES_REQUESTED** [review]

File: include/rest-request.php (hunk 1 above)
Comment (Screenfeed) [reply include/rest-request.php]: Done in a6700ee.

**Pull request merged**